src/components/select/simple-select/simple-select.test.js

```diff
@@ -0,0 +1,1022 @@
+        CypressMountWithProviders(
+          <SimpleSelectComponent labelInline labelAlign={alignment} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "-webkit-box-pack", cssProp)
+          .and("have.css", "justify-content", `flex-${cssProp}`);
+      }
+    );
+
+    it.each([
+      ["10", "90", "135px", "1215px"],
+      ["30", "70", "405px", "945px"],
+      ["80", "20", "1080px", "270px"],
+    ])(
+      "should use %s as labelWidth, %s as inputWidth and render it with correct label and input width ratios",
+      (label, input, labelRatio, inputRatio) => {
+        CypressMountWithProviders(
+          <SimpleSelectComponent
+            labelInline
+            labelWidth={label}
+            inputWidth={input}
+          />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "width", labelRatio);
+
+        getDataElementByValue("input")
+          .parent()
+          .should("have.css", "width", inputRatio);
+      }
+    );
+
+    it("should open the list with mouse click on Select input", () => {
+      CypressMountWithProviders(<SimpleSelectComponent />);
+
+      selectText().click();
+      commonDataElementInputPreview().should(
+        "have.attr",
+        "aria-expanded",
+        "true"
+      );
+      selectList().should("be.visible");
+    });
+
+    it("should open the list with mouse click on dropdown button", () => {
+      CypressMountWithProviders(<SimpleSelectComponent />);
+
+      dropdownButton().click();
+      commonDataElementInputPreview().should(
+        "have.attr",
+        "aria-expanded",
+        "true"
+      );
+      selectList().should("be.visible");
+    });
+
+    it("should close the list with the Tab key", () => {
+      CypressMountWithProviders(<SimpleSelectComponent />);
+
+      selectText().click();
+      commonDataElementInputPreview().should(
+        "have.attr",
+        "aria-expanded",
+        "true"
+      );
+      selectList().should("be.visible");
+      selectInput().trigger("keydown", { ...keyCode("Tab"), force: true });
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it("should close the list with the Esc key", () => {
+      CypressMountWithProviders(<SimpleSelectComponent />);
+
+      selectText().click();
+      commonDataElementInputPreview().should(
+        "have.attr",
+        "aria-expanded",
+        "true"
+      );
+      selectList().should("be.visible");
+      selectText().trigger("keydown", { ...keyCode("Esc") });
+      selectInput().should("have.attr", "aria-expanded", "false");
+      selectList().should("not.exist");
+    });
+
+    it("should close the list by clicking out of the component", () => {
+      CypressMountWithProviders(<SimpleSelectComponent />);
+
+      selectText().click();
+      commonDataElementInputPreview().should(
```

Comment:
same here

Reply:
Removed. I think I'd lost the will to live at this point and was just copying and pasting the main chunks of code but not removing what wasn't needed for each subsequent case :)